Sources/CorePayments/Networking/HTTP.swift

```diff
@@ -33,13 +33,13 @@ class HTTP {
             throw APIClientError.invalidURLResponseError
         }
         
-        if cachingEnabled {
-            let cachedURLResponse = CachedURLResponse(response: response, data: data)
-            urlCache.storeCachedResponse(cachedURLResponse, for: urlRequest)
-        }
-        
         switch response.statusCode {
         case 200..<300:
+            if cachingEnabled {
+                let cachedURLResponse = CachedURLResponse(response: response, data: data)
+                urlCache.storeCachedResponse(cachedURLResponse, for: urlRequest)
+            }
+            
             let decodedData = try decoder.decode(T.self, from: data)
```

Comment:
I think I should decode on top under line 37. Right now we are just screening for bad status codes before caching, not decoding error. Wdyt @scannillo @jaxdesmarais

Reply:
I also saw in the CardClient that we are not dealing with all errors like decoding errors. In approveOrder function, we don't send notifyFailure or analytic event in the general catch block and in getOrderInfo only catches CoreSDKError. Is there a reason for this?